### PR TITLE
Adding recipes for "GMat*" material models

### PR DESCRIPTION
**Pull request by tdegeus**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by tdegeus** [comment]
Note that the recipes are virtually identical. They differ only in their name, version, summary, and repository.

**Comment by tdegeus** [comment]
@conda-forge/help-c-cpp Ready for review.

Note that the recipes are virtually identical. They differ only in their name, version, summary, and repository.

File: recipes/python-gmatnonlinearelastic/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+source:
+  url: https://github.com/tdegeus/GMatNonLinearElastic/archive/v{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  skip: True  # [win]
+  script:
+    - python setup.py build
```
Comment (chrisburr) [inline recipes/python-gmatnonlinearelastic/meta.yaml]: Have you tried using pip?

Comment (tdegeus) [reply recipes/python-gmatnonlinearelastic/meta.yaml]: @chrisburr I successfully switched to pip, thanks!

**Pull request merged**